### ISSUE TITLE
Error with random_forest predict on Spark: Column `predicted_label` doesn't exist.

### ISSUE THREAD

**Issue by eamoncaddigan** [tidymodels/parsnip#365]
I just figured out a fix for a weird error running `predict` on a `rand_forest` classification model fit to a binary classification problem using the spark engine. When my outcome labels were encoded as a logical vector, `predict` would throw the following error:

> Error : Can't subset columns that don't exist.
> ✖ Column `predicted_label` doesn't exist.

When the outcome label is a factor, everything works fine!

I never used the name `predicted_label` for a column, so I'm guessing it was added along the way to handle the logical outcome label. I don't know if this is unique to parsnip or if this is coming from `sparklyr::ml_random_forest` or even Spark itself, but I don't think this behavior is documented anywhere (at least when I was trying to make my code work). So while I'm not sure that this counts as a _bug_, I am hoping that nobody else has to waste hours figuring this out. 😃 

tl;dr: when doing binary classification with random forests on Spark with parsnip, make sure your outcome is a factor.

**Comment by topepo** [comment]
Thanks for the note
